src/main/kotlin/com/hibob/academy/service/PetService.kt

```diff
@@ -1,10 +1,30 @@
 package com.hibob.academy.service
 
+import com.hibob.academy.dao.OwnerDao
 import com.hibob.academy.dao.PetDao
+import com.hibob.academy.entity.Owner
+import com.hibob.academy.entity.Pet
+import jakarta.ws.rs.BadRequestException
 import org.springframework.stereotype.Service
 
 @Service
 class PetService(
-    private val petDao: PetDao
+    private val petDao: PetDao,
 ) {
+
+    fun assignOwnerToPet(petId: Long, ownerId: Long) : Boolean {
+
+        return petDao.assignOwnerToPet(petId, ownerId)
+    }
+
+    fun getOwnerByPetId(petId: Long): Owner? {
```

Comment:
Not sure about the logic here
When do we get null from the dao? is it ok or should error be thrown?
I addition this code can be simplified

Same for the tests checking this in the PetDaoService